atest/testdata/keywords/type_conversion/python_annotations.robot

```diff
@@ -99,6 +99,14 @@ String None is converted to None object
     Dictionary
     Set
 
+Enum
+    Enum          BAR                 Foo.BAR
+
+Invalid Enum
+    Run Keyword And Expect Error
+    ...    ValueError: Argument 'argument' cannot be converted to Foo, got 'foobar'.
+    ...    Enum    foobar
+
 *** Keywords ***
 Conversion Should Fail
     [Arguments]    ${kw}    ${arg}
```

Comment:
You ought to be able to add new argument `${type}=${kw.lower()}` here and use this keyword also with the new `Invalid Enum` test like

    Conversin Should Fail    Enum    foobar    type=Foo

or with `[Template]` if you want to test multiple values like existing tests do.